src/components/TimeBasedLineChart.js

```diff
@@ -42,713 +255,294 @@ class TimeBasedLineChart extends React.Component {
-        .attr('x2', this._getXScaleValue(currentDate.unix()))
-        .attr('y2', this._getYScaleValue(belowValue))
-        .style('stroke-dasharray', dash)
-        .style('stroke-width', 3)
-        .style('stroke', '#C93030')
-        .style('fill', 'none');
-    }
+    chart.selectAll('.hovered-data-point-date-text')
+      .style(styles.dateTooltipText);
 
-    //Circle on line
-    dots.append('circle')
-      .attr('class', 'mx-time-based-line-chart-hover-dot')
-      .attr('cx', this._getXScaleValue(currentDate.unix()))
-      .attr('cy', this._getYScaleValue(hoveredData.value))
-      .attr('r', 3)
-      .style('fill', '#fff')
-      .style('stroke', this.props.lineColor)
-      .style('stroke-width', '2px');
-  }
+    // Style rest of chart elements
+    chart.selectAll('text').style(styles.text);
+    chart.selectAll('.domain').style(styles.domain);
+    chart.selectAll('.y-grid-line .tick').style('stroke', d => {
+      return d === 0 ? StyleConstants.Colors.CHARCOAL : StyleConstants.Colors.FOG;
+    })
+    .style('stroke-dasharray', d => {
+      return d === 0 ? 'none' : '4,4';
+    });
+  },
 
-  _renderTooltip () {
-    if (this.props.showTooltips && this.state.tooltipPosition) {
-      const hoveredData = this.state.hoveredData;
-      const position = this.state.tooltipPosition;
+  // Render functions
+  _renderHoveredDataPointDetails () {
+    if (this.props.hoveredDataPointDetails && this.state.hoveredDataPoint) {
+      return this.props.hoveredDataPointDetails.map((item, index) => {
+        const value = this.state.hoveredDataPoint[item.key];
 
-      if (this.props.children) {
         return (
-          <div className='mx-time-based-line-chart-tool-tip-wrapper' style={[styles.tooltipWrapper, position]}>
-            {this.props.children}
+          <div key={'details-' + index} style={styles.hoveredDataPointDetail}>
+            <div style={styles.hoveredDataPointLabel}>
+              {item.label}
+            </div>
+            <div style={styles.hoveredDataPointValue}>
+              {value || value === 0 ? this._getFormattedValue(value, item.type, item.format) : 'N/A'}
+            </div>
           </div>
         );
-      }
-
-      return (
-        <div style={[styles.tooltipWrapper, position]}>
-          <div style={styles.defaultToolTip}>
-            {hoveredData.value}
-          </div>
-        </div>
-      );
+      });
     }
-  }
+  },
 
   render () {
+    const { breakPointDate, breakPointLabel, data, height, lineColor, margin, rangeType, width, zeroState, yAxisFormatter } = this.props;
+    const { adjustedHeight, adjustedWidth, hoveredDataPoint } = this.state;
+
     return (
-      <div className='mx-time-based-line-chart' style={[styles.component, { height: this.props.height + 'px', width: this.props.width + 'px' }]}>
+      <div className='mx-time-based-line-chart' style={[styles.component, { height: this.props.height, width: this.props.width }]}>
```

Comment:
One last change here
`{ height: height, width: width }`, no need for `this.props`